scripts/templateCombinator.ts

```diff
@@ -0,0 +1,465 @@
+                  ),
+                  undefined,
+                  [_.createIdentifier(`is${name}`), _.createStringLiteral(name)]
+                ),
+              ])
+            ),
+          ],
+          ts.NodeFlags.Const
+        )
+      ),
+    ],
+    _.createNodeArray,
+    nodes => printer.printList(ts.ListFormat.MultiLine, nodes, sourceFile),
+    Str.replace(/\/\*\*/gm, '\n/**')
+  )
+}
+
+const PackageJson = SC.make(S =>
+  S.struct({
+    version: S.string,
+  })
+)
+
+const decodePackageJson = SC.interpreter(D.Schemable)(PackageJson)
+
+const getPackageJson: Build<SC.TypeOf<typeof PackageJson>> = C =>
+  pipe(
+    C.readFile('package.json'),
+    TE.chainEitherK(
+      flow(
+        Json.parse,
+        E.mapLeft(() => new Error('Unable to parse package.json'))
+      )
+    ),
+    TE.chainEitherKW(
+      flow(
+        decodePackageJson.decode,
+        E.mapLeft(() => new Error('Package.json does not match expected shape'))
+      )
+    )
+  )
+
+const checkModuleUniqueness: (module: string) => Build<ReadonlyArray<string>> =
+  module => C =>
+    pipe(
+      TE.Do,
+      TE.apS(
+        'date',
+        pipe(C.readFiles('./src/date'), TE.map(RA.map(flow(Str.split('.'), RNEA.head))))
+      ),
+      TE.apS(
+        'number',
+        pipe(C.readFiles('./src/number'), TE.map(RA.map(flow(Str.split('.'), RNEA.head))))
+      ),
+      TE.apS(
+        'string',
+        pipe(C.readFiles('./src/string'), TE.map(RA.map(flow(Str.split('.'), RNEA.head))))
+      ),
+      TE.map(({ date, number, string }) => [...date, ...number, ...string]),
+      TE.filterOrElse(
+        flow(
+          RA.findFirst(
+            existingModule => Str.toLowerCase(existingModule) === Str.toLowerCase(module)
+          ),
+          O.isNone
+        ),
+        () => new Error(`Module ${module} already exists`)
+      )
+    )
+
+const writeToDisk: (path: string) => (contents: string) => Build<void> =
+  path => contents => C =>
+    C.writeFile(path, contents)
+
+const format: Build<void> = C => C.exec('yarn format')
+
+const main: Build<void> = pipe(
+  process.argv,
+  RA.dropLeft(2),
+  RTE.fromPredicate(
+    (args): args is [string, string] => args.length === 2,
+    () =>
+      new Error(
+        'Must provide primitive type (string | number), and capitalized module name (SafeDate)'
+      )
+  ),
+  RTE.filterOrElse(
+    (args): args is ['string' | 'number', string] =>
+      args[0] === 'string' || args[0] === 'number',
+    () => new Error('Primitive must be one of "string", or "number"')
+  ),
+  RTE.filterOrElse(
+    ([, module]) => /^[A-Z]/g.test(module),
+    () => new Error('Module name must be capitalized')
+  ),
+  RTE.bindTo('args'),
+  RTE.chainFirstIOK(() => Cons.log("Checking that module doesn't already exist...")),
+  RTE.chainFirst(({ args: [, module] }) => checkModuleUniqueness(module)),
+  RTE.chainFirstIOK(() => Cons.log('Getting package version...')),
+  RTE.bind('packageVersion', () => getPackageJson),
```

Comment:
Most nitpick ever.. ⛳ 

```suggestion
  RTE.apSW('packageVersion', getPackageJson)
```

Reply:
No, you're totally right, this is an apS moment